x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,437 @@
+  });
+
+  describe('actions', () => {
+    it('setRoleMappingsData', () => {
+      RoleMappingsLogic.actions.setRoleMappingsData(mappingsServerProps);
+
+      expect(RoleMappingsLogic.values.roleMappings).toEqual([asRoleMapping]);
+      expect(RoleMappingsLogic.values.dataLoading).toEqual(false);
+      expect(RoleMappingsLogic.values.multipleAuthProvidersConfig).toEqual(true);
+    });
+
+    describe('setRoleMappingData', () => {
+      it('sets data correctly', () => {
+        RoleMappingsLogic.actions.setRoleMappingData(mappingServerProps);
+
+        expect(RoleMappingsLogic.values.roleMapping).toEqual(asRoleMapping);
+        expect(RoleMappingsLogic.values.dataLoading).toEqual(false);
+        expect(RoleMappingsLogic.values.attributes).toEqual(mappingServerProps.attributes);
+        expect(RoleMappingsLogic.values.availableEngines).toEqual(
+          mappingServerProps.availableEngines
+        );
+        expect(RoleMappingsLogic.values.accessAllEngines).toEqual(true);
+        expect(RoleMappingsLogic.values.elasticsearchRoles).toEqual(
+          mappingServerProps.elasticsearchRoles
+        );
+        expect(RoleMappingsLogic.values.selectedEngines).toEqual(
+          new Set(engines.map((e) => e.name))
+        );
+      });
+
+      it('sets default group with new role mapping', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          roleMapping: undefined,
+        });
+
+        expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set());
+      });
+    });
+
+    it('handleRoleChange', () => {
+      RoleMappingsLogic.actions.handleRoleChange('dev');
+
+      expect(RoleMappingsLogic.values.roleType).toEqual('dev');
+    });
+
+    it('handleEngineSelectionChange', () => {
+      const engine = engines[0];
+      const otherEngine = engines[1];
+      RoleMappingsLogic.actions.setRoleMappingData({
+        ...mappingServerProps,
+        roleMapping: {
+          ...asRoleMapping,
+          engines: [engine, otherEngine],
+        },
+      });
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, true);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(
+        new Set([engine.name, otherEngine.name])
+      );
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, false);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set([engine.name]));
+    });
+
+    it('handleAccessAllEnginesChange', () => {
+      RoleMappingsLogic.actions.handleAccessAllEnginesChange();
+
+      expect(RoleMappingsLogic.values.accessAllEngines).toEqual(false);
+    });
+
+    describe('handleAttributeSelectorChange', () => {
+      const elasticsearchRoles = ['foo', 'bar'];
+
+      it('sets values correctly', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          elasticsearchRoles,
+        });
```

Comment:
I think you should prefer to use the mount helper to set initial data for a test.

```
mount({
  attributeValue: /*whatever this should be initially*/,
  attributeName: /*whatever this should be initially*/
})
```

It server as a better "diff" when looking at a test ... meaning it's easy to see what a reducer should be before the action runs, and what it should be after it runs. 

It also keeps the test isolated ... this test needs to test `handleAttributeSelectorChange`, not `setRoleMappingData`.

Reply:
Addressed here (https://github.com/elastic/kibana/pull/94461/commits/eb1298a3bd9c43e9251d110e70c290dce6be9966)